frame/im-online/src/mock.rs

```diff
@@ -90,6 +101,31 @@ pub fn new_test_ext() -> sp_io::TestExternalities {
 	t.into()
 }
 
+#[derive(Clone, Debug, Eq, PartialEq, Encode, Decode)]
+pub struct PublicWrapper(Public);
+
+impl IdentifyAccount for PublicWrapper {
+	type AccountId = UintAuthorityId;
+	fn into_account(self) -> Self::AccountId {
+		let bytes = self.0.to_vec();
+		let mut truncated = [0u8; 8];
+		truncated.copy_from_slice(bytes.as_slice());
+		UintAuthorityId(u64::from_be_bytes(truncated))
```

Comment:
why not keep accountId as u64 and return `self.0` here? would probably keep things simpler as working with u64 is easier.

Reply:
I followed your suggestion. I initially had to change it all as i wanted AuthorityId to implement `RuntimeAppPublic` as required by the offchain module for `OffchainAuthorityId`  but since i implemented the `PublicWrapper`, the change of `AuthorityId` was no longer needed. Reverted the changes.